providers/implementations/kdfs/scrypt.c

```diff
@@ -507,10 +507,6 @@ static int scrypt_alg(const char *pass, size_t passlen,
         return 0;
     }
 
-    /* Check that the maximum memory doesn't exceed a size_t limits */
-    if (maxmem > SIZE_MAX)
-        maxmem = SIZE_MAX;
-
     if (Blen + Vlen > maxmem) {
```

Comment:
This `if` will also have an overflow risk as will the `N + 2` above.
I'm not sure they are under an attacker's influence or not.